### PR TITLE
sendrawtx API

### PR DESCRIPTION
**Pull request by mbg033**
- integrated cryptonode libraries (dynamic linkage only)
- sendrawtx API (call to cryptonode)
- JSON-RPC macros (from https://github.com/graft-project/graft-ng/pull/20)

### CODE REVIEW

File: test/cryptonode_handlers_test.cpp [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    }
+
+    virtual void SetUp() override
+    { }
+    virtual void TearDown() override
+    { }
+
+    GraftServer server;
+    Manager   * manager;
+    std::thread server_thread;
```
Comment (AlexanderSuprunenko) [inline test/cryptonode_handlers_test.cpp]: Why not?
Manager  manager; //not a pointer
GraftServer server;
Please, pay attention to the order, in this case server dtor will be called before manager call. And you can place server_thread.join into ~CryptonodeHandlersTest().
Also note, according to the mongoose, mg_mgr_free should close all open connections, including listening.
Also, as I know CryptonodeHandlersTest object are created for each TEST_F, and the order of TEST_Fs is not specified.

File: src/graft_manager.cpp [hunk 2]
```diff
@@ -412,13 +429,14 @@ void GraftServer::ev_handler_http(mg_connection *client, int ev, void *ev_data)
 
         struct http_message *hm = (struct http_message *) ev_data;
         std::string uri(hm->uri.p, hm->uri.len);
+        // TODO: why this is hardcoded ?
```
[the remark this answers is not in the document]
Comment (mbg033) [reply src/graft_manager.cpp]: There should be C++ interface to stop the server

[the remark this answers is not in the document]
Comment (mbg033) [reply src/graft_manager.cpp]: There should be C++ interface to stop the server, not as hardcoded endpoint

File: test/cryptonode_handlers_test.cpp (hunk 1 above)
Comment (mbg033) [reply test/cryptonode_handlers_test.cpp]: > And you can place server_thread.join into ~CryptonodeHandlersTest().
intent was **explicitly** wait for server thread in the test code

File: include/graft_manager.h [hunk 3]
```diff
@@ -152,7 +156,7 @@ class Manager
 
     ServerOpts m_sopts;
 public:
-    bool exit = false;
+    std::atomic_bool m_exit {false};
```
[the remark this answers is not in the document]
Comment (mbg033) [reply include/graft_manager.h]: bump

**Pull request merged**